sw/airborne/firmwares/fixedwing/stabilization/stabilization_adaptive.c

```diff
@@ -520,3 +596,103 @@ inline static void h_ctl_pitch_loop(void)
   h_ctl_elevator_setpoint = TRIM_PPRZ(cmd);
 }
 
+/* yaw damper */
+#if H_CTL_YAW_LOOP
+inline static void h_ctl_yaw_loop(void)
+{
+ 
+#if H_CTL_YAW_TRIM_BY
+  // Actual Acceleration from IMU:
+ #ifndef SITL
+  struct Int32Vect3 accel_meas_body;
+  struct Int32RMat *body_to_imu_rmat = orientationGetRMat_i(&imu.body_to_imu);
+  int32_rmat_transp_vmult(&accel_meas_body, body_to_imu_rmat, &imu.accel);
```

Comment:
IMU data (at least the `accel` here) should not be accessed directly in control loops.
You have the solution to get the NedAccel and NedToBodyRMat from state interface, or bind to sensor message like ins/ahrs filters.
Here, I suggest solution 1.